File: java/ankit_kanyal/devnest_problems/FloodFill.java
// link- https://devsnest.in/algo-challenges/flood-fill

static int[][] solve(int[][] image, int sr, int sc, int color){
//CODE HERE 

    if (image[sr][sc] == color) return image;
        fill(image, sr, sc, image[sr][sc], color);
        return image;
}

public static void fill(int [][] image,int sr,int sc,int color,int newColor){

    if(sr<0 || sr>=image.length || sc<0 || sc>=image[0].length || image[sr][sc]!=color) return ;

    image[sr][sc]=newColor;
    fill(image,sr+1,sc,color,newColor);
    fill(image,sr,sc+1,color,newColor);
    fill(image,sr-1,sc,color,newColor);
    fill(image,sr,sc-1,color,newColor);
}

File: python/vedant_bhatnagar/DevsnestProblems_THA/Phase 1/FloodFill.py
def solve(image , sr, sc, color):
    # CODE HERE
    original = image[sr][sc]
    if original == color:
        return image

    image[sr][sc] = color
    if sr>0 and image[sr-1][sc] == original:
        solve(image, sr-1,sc,color)
    if sr<len(image)-1 and image[sr+1][sc] == original:
        solve(image, sr+1,sc,color)
    if sc>0 and image[sr][sc-1] == original:
        solve(image, sr,sc-1,color)
    if sc < len(image[0])-1 and image[sr][sc+1] == original:
        solve(image, sr,sc+1,color)
    return image